chain-impl-mockchain/src/transaction/witness.rs

```diff
@@ -56,8 +56,13 @@ impl std::fmt::Display for Witness {
 pub struct WitnessUtxoData(Vec<u8>);
 
 impl WitnessUtxoData {
-    pub fn new(block0: &HeaderId, transaction_id: &TransactionSignDataHash) -> Self {
+    pub fn new(block0: &HeaderId, transaction_id: &TransactionSignDataHash, is_legacy: bool) -> Self {
         let mut v = Vec::with_capacity(65);
+        let tag = match is_legacy {
+            true => WITNESS_TAG_OLDUTXO,
+            false => WITNESS_TAG_UTXO,
+        };
+        v.push(tag);
```

Comment:
This is a breaking cryptographic change, so this need some coordination with others here.